check_diff/src/lib.rs

```diff
@@ -56,3 +128,83 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
     );
     return Ok(());
 }
+
+// Compiles and produces two rustfmt binaries.
+// One for the current master, and another for the feature branch
+// Parameters:
+// dest: Directory where rustfmt will be cloned
+pub fn compile_rustfmt(
+    dest: &Path,
+    remote_repo_url: String,
+    feature_branch: String,
+    commit_hash: Option<String>,
+) -> Result<Command, CheckDiffError> {
+    const RUSTFMT_REPO: &str = "https://github.com/rust-lang/rustfmt.git";
+
+    let clone_git_result = clone_git_repo(RUSTFMT_REPO, dest);
+    if clone_git_result.is_err() {
+        return Err(CheckDiffError::FailedGit(clone_git_result.err().unwrap()));
+    }
+
+    let git_remote_add_result = git_remote_add(remote_repo_url.as_str());
+    if git_remote_add_result.is_err() {
+        return Err(CheckDiffError::FailedGit(
+            git_remote_add_result.err().unwrap(),
+        ));
+    }
+
+    let git_fetch_result = git_fetch(feature_branch.as_str());
+    if git_fetch_result.is_err() {
+        return Err(CheckDiffError::FailedGit(git_fetch_result.err().unwrap()));
+    }
+
+    let cargo_version = env!("CARGO_PKG_VERSION");
+    info!("Compiling with {}", cargo_version);
+
+    //Because we're building standalone binaries we need to set `LD_LIBRARY_PATH` so each
+    // binary can find it's runtime dependencies.
+    // See https://github.com/rust-lang/rustfmt/issues/5675
+    // This will prepend the `LD_LIBRARY_PATH` for the master rustfmt binary
+    let Ok(command) = Command::new("rustc").args(["--print", "sysroot"]).output() else {
+        return Err(CheckDiffError::FailedCommand(
+            "Error getting sysroot".to_string(),
+        ));
+    };
+
+    let Ok(sysroot) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting sysroot to string".to_string(),
+        ));
+    };
+
+    let _ld_lib_path = format!("{}/lib", sysroot.trim_end());
+    info!("Building rustfmt from source");
+    let Ok(_) = Command::new("cargo")
```

Comment:
I'm not sure whether we are suppose to use `ld_lib_path` here as a `.env` argument

Reply:
Yeah, you'd want to do something like that.

Reply:
Okay, maybe I confused you. We don't need the `LD_LIBRARY_PATH` when we compile rustfmt since `cargo` will set that correctly for us. We need to know the `LD_LIBRARY_PATH` so that we can call the rustfmt binary.